doc/api/packages.md

```diff
@@ -819,10 +747,213 @@ conditional exports for consumers could be to add an export, e.g.
 }
 ```
 
+## Node.js `package.json` field definitions
+
+This section describes the fields used by the Node.js runtime. Other tools (such
+as [npm](https://docs.npmjs.com/creating-a-package-json-file)) may use
+additional fields which are ignored by Node.js and not documented here.
+
+### `"name"`
+<!-- YAML
+added:
+  - v12.16.0
+  - v13.1.0
+changes:
+  - version:
+    - v12.16.0
+    - v13.6.0
+    pr-url: https://github.com/nodejs/node/pull/31002
+    description: Remove the `--experimental-resolve-self` option.
+-->
+
+```json
+{
+  "name": "package-name"
+}
+```
+
+* Type: {string}
+
+The `"name"` field defines your package’s name. Node.js doesn't apply any
+restriction on the name field, although the field is ignored if it is not a
+string or an empty string. Other tools (such as
+[npm](https://docs.npmjs.com/files/package.json#name)) may apply other
+restrictions on this field.
+
+The `"name"` field can be used in addition to the [`"exports"`][] field to
+[self-reference][] a package using its name.
+
+### `"type"`
+<!-- YAML
+added: v12.0.0
+changes:
+  - version:
+    - v12.17.0
+    - v13.2.0
+    pr-url: https://github.com/nodejs/node/pull/29866
+    description: Unflag `--experimental-modules`.
+-->
+
+* Type: {string}
+
+The `"type"` field defines the module format `.js` files should be treated
+within a particular `package.json` file’s package scope.
+
+Files ending with `.js` will be loaded as ES modules when the nearest parent
+`package.json` file contains a top-level field `"type"` with a value of
+`"module"`.
+
+The nearest parent `package.json` is defined as the first `package.json` found
+when searching in the current folder, that folder’s parent, and so on up
+until a node_modules folder or the volume root is reached.
+
+```json
+// package.json
+{
+  "type": "module"
+}
+```
+
+```console
+# In same folder as above package.json
+node my-app.js # Runs as ES module
+```
+
+If there is no package.json scope found, or if the nearest parent `package.json`
+lacks a `"type"` field, or contains `"type": "commonjs"`, or any other
+unsupported value, `.js` files are treated as CommonJS.
```

Comment:
```suggestion
If the nearest parent `package.json` lacks a `"type"` field, or contains	
`"type": "commonjs"`, `.js` files are treated as [CommonJS][]. If the volume	
root is reached and no `package.json` is found, Node.js defers to the default, a	
`package.json` with no `"type"` field.

`import` statements of `.js` files are treated as ES modules if the nearest	
parent `package.json` contains `"type": "module"`.
```

Let's keep the original text whenever possible please, it's gone through a lot of vetting already. "If there is no package.json scope found"? There's always a package scope. And let's leave out anything about unsupported values; there may be new types added in the future.

The second paragraph appears to have been lost in the reorganization, but it's important because some people assume that `import` always treats the target as ESM, but it doesn't.

Reply:
This was at my suggestion to treat this section of the docs more like a comprehensive reference definition, while treating the top section of this page more like guide material. I actively encouraged refactoring as part of this process, just like refactoring when adding a feature I think it is natural and we shouldn't split out rewordings separately or risk not doing them at all.

Reply:
> This was at my suggestion

Fair enough. I don't think this particular rewording is an improvement, though, and it covers the same content as the previous language.

Reply:
The point was to make these definitions the comprehensive ones - as in to ensure concepts like the root scope are covered.

Reply:
Then allow the top-of-the-page guide documentaiton to be able to skip these kinds of details rather.

Reply:
> allow the top-of-the-page guide documentaiton to be able to skip these kinds of details rather.

This is in the bottom of the page . . . ? Are you opposed to my suggested change?

Reply:
The only thing I would add is that I prefer to say the default treatment is commonjs than "no type" being a thing.

Otherwise the rewording seems fine to me.